src/mongocxx/test/database.cpp

```diff
@@ -71,6 +71,20 @@ TEST_CASE("A database", "[database]") {
         REQUIRE(destroy_called);
     }
 
+    SECTION("is dropped") {
+        bool drop_called = false;
+
+        database_drop->interpose([&](mongoc_database_t* database, bson_error_t* error) { 
+            drop_called = true; 
+            return true;
+        });
+
+        database database = mongo_client["database"];
+        REQUIRE(!drop_called);
+        database.drop();
+        REQUIRE(drop_called);
+    }
```

Comment:
can you add a negative test showing that the call will throw if database_drop returns false. See REQUIRE_THROWS